### DISCUSSION TITLE
SVG Issues When Converting To PNG

### DISCUSSION THREAD

**Discussion by mackenly** [simple-icons/simple-icons#7971] [General]
Hi, I'm the maintainer of the Stream Deck Simply Icons 3rd party extension. I've been running into a bug where certain icons don't display correctly. They look deformed. I first thought it must be a Stream Deck issue, so tried converting the SVGs to PNGs using a Python library, but after converting them, they had the same issues. After that, I tried one of the icons (Home Assistant) in an [SVG optimizer/validator](https://www.svgviewer.dev/), which seemed to fix the issue.

Is there a known issue related to certain icons looking deformed that others have experienced as well? I've eliminated the stream deck as being the issue since the same problem happens when converting just on my machine.

Here's a screengrab from an issue in the simple icons stream deck repo to illustrate what I'm referring to: https://github.com/mackenly/simple-icons-stream-deck/issues/1
![image](https://user-images.githubusercontent.com/53151058/196269994-71b34514-ed2e-4e14-8131-8350421a480a.png)

**Comment by LitoMore** [comment]
I think it's a bug in the Python library you use.

**Reply by mackenly** [reply]
A possibility, however, that wouldn't account for the same issue happening when you directly add an SVG from simple icons to a stream deck via stream deck's software (unless they're somewhere using the same library).

I'm thinking the solution is probably finding a good python library that can process it in a way that plays nice.

I don't know enough about how SVGs work to understand what could be causing a particular image to work one place but not another, wondering if there are different implementation approaches for SVGs. Anyways, thanks for the thoughts! I'll have to look a bit deeper.

**Reply by LitoMore** [reply]
As we mentioned in our [contributing doc](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#3-optimize-the-icon). We require contributors to optimize the SVG icon for a smaller file size. We usually use SVGO or SVGOMG for optimizing.

It might be a bug in the Python library that cannot analyze the optimized content correctly.

I don't know too much about the Stream Deck. You could find a JavaScript tool instead the Python one if there is no suitable Python tool and the Stream Deck Plugin SDK supports JavaScript.

**Comment by 8KCool** [comment]
It might be a bug in the python so you can not analyze it.
I also want to know about this.
